File: Scripts/Models/Habit.js
const Habit = class {
    constructor (
        id
        , name
        , removed
        , startTime
        , endTime
        , days
        , lastLogTime
    ) {
        this.id = id
        this.name = name
        this.removed = removed
        this.startTime = startTime
        this.endTime = endTime
        this.days = days
        this.lastLogTime = lastLogTime
    }
}

module.exports = { Habit }

File: ProTracker.Backend/ProTracker.Models/Habit.cs
namespace ProTracker.Models;

public class Habit
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool Removed { get; set; }
    public long StartTime { get; set; }
    public long? EndTime { get; set; }
    public int Days { get; set; }

    // Navigation properties
    public ICollection<HabitLog> HabitLogs { get; set; } = new List<HabitLog>();
}
